pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -797,7 +795,9 @@ def process_circuits(
             else:
                 c0, ppcirc_rep = circ, None
             results_selection = []
-            for name, count in Counter(bit.reg_name for bit in c0.bits).items():
+            for name, count in Counter(
+                bit.reg_name for bit in c0.bits if not _is_scratch(bit)
```

Comment:
Does this always ignore scratch bits, even if they are not erased?

Reply:
It removes them all, which I assumed would be what we wanted. Can there be scratch bits that are actually used?

Reply:
Yes I guess we cannot assume this...

Reply:
Fixed I think. Integration tests running here: https://github.com/CQCL/pytket-quantinuum/actions/runs/6039214979 .